Feat/add azure openai (#301)
mckinsey/vizro: vizro-ai/docs/pages/user_guides/model_config.md

which would be a good starting point for most users. `gpt-4` models are more powerful than `gpt-3` models, e.g. they allow for more tokens per request. You can refer to these models' [capabilities](https://platform.openai.com/docs/models/overview)
and [pricing](https://openai.com/pricing) for more information.

> ### Azure OpenAI models

Reviewer: Again I'm curious how general this solution is. It feels like it should work for more than just Azure OpenAI, which is just one example?

Author: It could work for any model in theory,  but it should not be allowed in practice. 
- In code it will work without throwing an error as langchain allows to add any models here(google ones, cohere, anything that langchain supports).
- but for supported LLM, we only have gpt3.5 and gpt4 with openai right now since we are using `openai function calling` in all LLM calls. No other models are supported at this moment. 

tbh with regards to this, it would probably be better to use the property.setter than doing this, it leads to a bit different conclusion from our discussion last time if we want to be more vigilant in terms of what should be allowed here in `vizro-ai`

which would be a good starting point for most users. `gpt-4` models are more powerful than `gpt-3` models, e.g. they allow for more tokens per request. You can refer to these models' [capabilities](https://platform.openai.com/docs/models/overview)
and [pricing](https://openai.com/pricing) for more information.

### Azure OpenAI models

> The setup below allows you to integrate Azure's OpenAI models with VizroAI, by simply updating the `llm_to_use` attribute in `VizroAI`.

Reviewer 2: ```suggestion
The setup described below enables the integration of Azure's OpenAI models with VizroAI. You can achieve this by simply updating the `llm_to_use` attribute in `VizroAI`.
```

> To set up Azure OpenAI with Langchain, you'll configure the AzureOpenAI instance by specifying your deployment name and model name.

Reviewer 2: ```suggestion
To set up Azure OpenAI with Langchain, you'll need to configure the AzureOpenAI instance by specifying your deployment name and model name.
```

> Environment variables for API configuration, such as `OPENAI_API_TYPE`, `OPENAI_API_VERSION`, `OPENAI_API_BASE` and `OPENAI_API_KEY`.

Reviewer 2: Not sure I understand this? Is this part of the sentence above? :D or is there a part missing?

"Environment variables for API configuration, such as `OPENAI_API_TYPE`, `OPENAI_API_VERSION`, `OPENAI_API_BASE` and `OPENAI_API_KEY`."

Author: ah yes! thank you for catching this. fixed now.

Authentication can be done via an API key directly or through Azure Active Directory (AAD) for enhanced security.

> For a detailed walkthrough, refer to the Langchain documentation directly.
> https://python.langchain.com/docs/integrations/llms/azure_openai

Reviewer 2: ```suggestion
For a detailed walkthrough, refer to the [Langchain documentation](https://python.langchain.com/docs/integrations/llms/azure_openai) directly.
```

> Here is an example of how to set AzureOpenAI model in `vizro_ai`.

Reviewer 2: ```suggestion
Here is an example of how to set the LLM model to be an AzureOpenAI model:
```
